### PR TITLE
feat: add subtitle to analytics widget

### PR DESCRIPTION
**Pull request by parkerjohn**
Is required for [#804](https://github.com/withthegrid/platform-client/pull/804)

Is required for [#887](https://github.com/withthegrid/platform/pull/887)

### CODE REVIEW

File: src/models/analytics-panel.ts [hunk 1]
```diff
@@ -59,6 +59,7 @@ const schema = Joi.object().keys({
   layout: layoutSchema.default([]),
   cards: Joi.array().items(Joi.object().keys({
     title: Joi.string().required().example('My widget'),
+    subTitle: Joi.string().allow('').optional(),
```
Comment (everhardt) [inline src/models/analytics-panel.ts:62]: * Shall we write it as `subtitle`? Wikipedia writes it as one word: https://en.wikipedia.org/wiki/Subtitles
* `.optional()` is not needed: if you do not specify `.required()` it is optional by default

File: src/models/analytics-panel.ts [hunk 2]
```diff
@@ -81,6 +82,7 @@ interface AnalyticsPanel {
   layout: AnalyticsPanelLayout;
   cards: {
     title: string;
+    subTitle: string;
```
Comment (everhardt) [inline src/models/analytics-panel.ts:85]: As it's optional, it should be `subtitle?: string`

File: src/routes/analytics/add-panel.ts [hunk 3]
```diff
@@ -32,6 +32,7 @@ const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
     layout: layoutSchema.default([]),
     cards: Joi.array().items(Joi.object().keys({
       title: Joi.string().required().example('My widget'),
+      subTitle: Joi.string().allow('').optional(),
```
Comment (everhardt) [inline src/routes/analytics/add-panel.ts:35]: It's missing from the TS interface above.

**Review by everhardt — CHANGES_REQUESTED** [review]

File: src/models/analytics-panel.ts (hunk 1 above)
Comment (yarsky-tgz) [inline src/models/analytics-panel.ts:62]: shouldn't be better to provide also any example?

Comment (yarsky-tgz) [inline src/models/analytics-panel.ts:62]: actually `.allow('')` is redundant, while field is optional

**Review by yarsky-tgz — CHANGES_REQUESTED** [review]
Having some comments

File: src/models/analytics-panel.ts (hunk 1 above)
Comment (everhardt) [reply src/models/analytics-panel.ts:62]: I don't think that's true:
image

Comment (yarsky-tgz) [reply src/models/analytics-panel.ts:62]: Oh. My bad, closing, sorry for disturbing.

Comment (parkerjohn) [reply src/models/analytics-panel.ts]: No idea what I was thinking there with the camelcase :)
Removed optional() everywhere

File: src/models/analytics-panel.ts (hunk 2 above)
Comment (parkerjohn) [reply src/models/analytics-panel.ts]: Fixed

File: src/models/analytics-panel.ts (hunk 1 above)
Comment (parkerjohn) [reply src/models/analytics-panel.ts]: Done

File: src/routes/analytics/add-panel.ts (hunk 3 above)
Comment (parkerjohn) [reply src/routes/analytics/add-panel.ts]: Added

**Review by everhardt — APPROVED** [review]
Ok! Will hold off merging until the platform-client PR is ready

**Review by yarsky-tgz — APPROVED** [review]

**Pull request merged**